src/lib.rs

```diff
@@ -37,39 +37,81 @@ mod macos;
 use macos as platform;
 
 use std::env;
-use std::fs::File;
+use std::ffi::OsStr;
+use std::fs::{self, File};
 use std::io::BufReader;
 use std::io::{Error, ErrorKind};
 use std::path::{Path, PathBuf};
 
 use pki_types::CertificateDer;
 
+const ENV_CERT_FILE: &str = "SSL_CERT_FILE";
+const ENV_CERT_DIR: &str = "SSL_CERT_DIR";
+
 /// Load root certificates found in the platform's native certificate store.
 ///
-/// If the SSL_CERT_FILE environment variable is set, certificates (in PEM
-/// format) are read from that file instead.
+/// ## Environment Variables
+///
+/// | Env. Var.      | Description                                                         |
+/// |----------------|---------------------------------------------------------------------|
+/// | SSL_CERT_FILE  | File containing an arbitrary number of certificates in PEM format.  |
+/// | SSL_CERT_DIR   | Directory utilizing the hierarchy used by OpenSSL's [c_rehash].     |
+///
+/// If **either** (or **both**) are set, certificates are only loaded from
+/// the locations specified via environment variables and not the platform-
+/// native certificate store.
 ///
-/// This function fails in a platform-specific way, expressed in a `std::io::Error`.
+/// ## Errors
+///
+/// This function fails in a platform-specific way, expressed in a [`std::io::Error`].
+///
+/// ## Caveats
 ///
 /// This function can be expensive: on some platforms it involves loading
 /// and parsing a ~300KB disk file.  It's therefore prudent to call
 /// this sparingly.
+///
+/// [c_rehash]: https://www.openssl.org/docs/manmaster/man1/c_rehash.html
 pub fn load_native_certs() -> Result<Vec<CertificateDer<'static>>, Error> {
-    load_certs_from_env().unwrap_or_else(platform::load_native_certs)
+    if let Some(certs) = load_certs_from_env()? {
+        return Ok(certs);
+    };
+    platform::load_native_certs()
 }
 
-const ENV_CERT_FILE: &str = "SSL_CERT_FILE";
-
-/// Returns None if SSL_CERT_FILE is not defined in the current environment.
+/// Returns certificates stored at SSL_CERT_FILE and/or SSL_CERT_DIR.
+///
+/// If neither is set, `None` is returned.
+///
+/// If SSL_CERT_FILE is defined, a file must exist at this path, and loading
+/// certificates from it must succeed. However, [`rustls_pemfile::certs()`]
+/// will try to skip parts of a file which are not part of a certificate, and the
+/// file may contain no certificates.
 ///
-/// If it is defined, it is always used, so it must be a path to an existing,
-/// accessible file from which certificates can be loaded successfully. However,
-/// [`rustls_pemfile::certs()`] will try to skip parts of a file which
-/// are not part of a certificate.
-fn load_certs_from_env() -> Option<Result<Vec<CertificateDer<'static>>, Error>> {
-    let cert_var_path = PathBuf::from(env::var_os(ENV_CERT_FILE)?);
+/// If SSL_CERT_DIR is defined, a directory must exist at this path, and all
+/// [hash files](`is_hash_file_name()`) contained in it must be loaded successfully,
```

Comment:
I think this link target doesn't work out when docs are built w/ `cargo doc --no-deps --document-private-items`, but this seems to work:

```
/// [hash files][is_hash_file_name] contained in it must be loaded successfully,
```

Reply:
When I wrote this, I wasn't sure about the syntax, so, I just ended up testing it and the link appeared to work just fine.

[Looking at the documentation](https://doc.rust-lang.org/rustdoc/write-documentation/linking-to-items-by-name.html#linking-to-items-by-name), there appear to be multiple options.

Reply:
Apologies, it does appear to render correctly as-is when I restested just now.